libkbfs/folder_branch_ops.go

```diff
@@ -537,6 +537,11 @@ func (fbo *folderBranchOps) setHeadLocked(
 		fbo.setLatestMergedRevisionLocked(ctx, lState, md.Revision, false)
 	}
 
+	if md.MergedStatus() == Unmerged && md.BID != fbo.bid {
```

Comment:
To be clearer (since re-reading this confused me) I meant a check that either the head and its successor are both merged, or they're both unmerged with the same branch ID, or that the head is merged and the successor is unmerged.